haskell-indexer-plugin/src/Haskell/Indexer/Plugin.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Haskell.Indexer.Plugin where
+
+import GhcPlugins hiding (($$))
+import TcRnTypes
+
+import TcRnMonad
+import TcRnDriver
+import GhcMonad
+import Data.IORef
+
+import Language.Haskell.Indexer.Backend.Ghc
+import Language.Haskell.Indexer.Backend.GhcEnv
+import Language.Haskell.Indexer.Translate
+import Language.Haskell.Indexer.Args
+
+import System.IO
+import System.FilePath
+import System.Directory
+
+plugin :: Plugin
+plugin = defaultPlugin
+          { renamedResultAction   = keepRenamedSource
+          , typeCheckResultAction = install }
+
+mkPath :: FilePath -> Module -> FilePath
+mkPath fp m
+  = fp </> (moduleNameString (moduleName m) ++ (show (moduleUnitId m)))
```

Comment:
Does `serve.sh` expect these to be called `*.entries`? It looks like either the script needs to be amended, or the plugin needs to change the names.

Reply:
It seems like `serve.sh` did expect them to be called `.entries`. I don't see any particular reason for it to expect that though and I'm not sure if it is a developed convention.

Reply:
@robinp ^ ?

Reply:
It is a convention, but not forced.

Reply:
I made it write `.entries` files anyway now as the filename is not important.